e2e/example_runner.go

```diff
@@ -57,19 +57,21 @@ func (f *ExamplesFramework) Create(fileName string) {
 	// try to serialize it into Unstructured or UnstructuredList
 	data, err := ioutil.ReadFile(fileName)
 	Expect(err).NotTo(HaveOccurred())
-	jsonData, err := utilyaml.ToJSON(data)
-	Expect(err).NotTo(HaveOccurred())
-	var ust runtime.Unstructured
-	err = runtime.DecodeInto(api.Codecs.UniversalDecoder(), jsonData, &ust)
-	if err != nil {
-		var ustList runtime.UnstructuredList
-		err = runtime.DecodeInto(api.Codecs.UniversalDecoder(), jsonData, &ustList)
+
+	utils.ForEachYamlDocument(data, func(doc []byte) {
```

Comment:
Thanks for pointing out. Maybe we can change the implementation later. I chose this method because it requires minimum changes to existing code

As for the tests - it is tested by flow e2e tests because they test not only that example finished deploying, but also what resources were created. Without this fix only the first deployment would be loaded and the tests would fail. Also considering that the code is part of the tests, it would be tests of tests